src/passes/OptimizeInstructions.cpp

```diff
@@ -1210,7 +1210,13 @@ struct OptimizeInstructions
   }
 
   // We can combine `or` operations, e.g.
-  //   (x > y) | (x == y)    ==>    x >= y
+  //   (x >  y) | (x == y)    ==>    x >= y
+  //   (x >= y) | (x == y)    ==>    x >= y
+  //   (x <  y) | (x == y)    ==>    x <= y
+  //   (x <= y) | (x == y)    ==>    x <= y
+  //   (x >  y) | (x <  y)    ==>    x != y
+  //   (x != y) | (x == y)    ==>    1
+  //   (x >= y) | (x <= y)    ==>    1
```

Comment:
How are the left-side `>=` expressions handled? I don't see those cases below.

Reply:
Added

Reply:
It looks like the tests didn't need updating, but I see a few examples in the tests with i32.ge_s on the left hand side. Were they somehow being handled already by previous code?

Reply:
My guess it swapped (canonize) by code earlier:
```c++
if (Properties::isSymmetric(binary)) {
  canonicalize(binary);
}
```
here: https://github.com/WebAssembly/binaryen/blob/master/src/passes/OptimizeInstructions.cpp#L348